incipyt/actions/git.py

```diff
@@ -0,0 +1,97 @@
+from incipyt import hooks
+
+from incipyt._internal import templates
+from incipyt._internal.dumpers import Requirement
+
+
+class Git:
+    """Action to add Git to :class:`incipyt.system.Hierarchy`."""
+
+    def __init__(self):
+        hooks.VCSIgnore.register(self._hook)
+
+    def add_to(self, hierarchy):
+        """Add git configuration to `hierarchy`.
+
+        Register git related project URLs:
+        - Repository: {Repository}
+        - Issue: {Repository}/issues
+        - Documentation: {Repository}/wiki
+
+        :param hierarchy: The actual hierarchy to update with git configuration.
+        :type hierarchy: :class:`incipyt.system.Hierarchy`
+        """
+        hook_url = hooks.ProjectURL(hierarchy)
+        hook_url("Repository", templates.Requires("{Repository}"))
+        hook_url("Issue", templates.Requires("{Repository}/issues"))
+        hook_url("Documentation", templates.Requires("{Repository}/wiki"))
+
+    def _hook(self, hierarchy, value):
+        gitignore = hierarchy.get_configuration(Requirement.make(".gitignore"))
+        gitignore[None] = [value]
+
+    def __str__(self):
+        return "git"
+
+    def pre(self, workon, environment):
+        """Run `git init`.
+
+        Also push {AUTHOR_NAME} and {AUTHOR_EMAIL} using `git config user.*`.
+
+        :param workon: Work-on folder.
+        :type workon: :class:`pathlib.Path`
+        :param environment: Environment used to do pre-action
+        :type environment: :class:`incipyt.system.Environment`
+        """
+        environment.run(
+            [
+                "git",
+                "init",
+                str(workon),
+            ]
+        )
```

Comment:
Honestly, it's not black here, it forces like that because I put a `,` after last element, if not it results to `environment.run(["git", "init", str(workon)])`.

Reply:
fixed